### PR TITLE
feat: Allow specifying multiple user ids for export

### LINKED ISSUE #8: Allow specifying multiple user IDs for export
Currently, you can only provide a single user ID to export via the `-userId <userId>` command line parameter. However, we have the need to export data for a small subset of users (e.g., 5-20 users).

We should support some kind of input, either via multiple strings on the command line or by providing a file of user IDs, where we can export data for multiple users at once. Data should be exported for each user in their own directory, named with their user ID.

### PR DESCRIPTION
**Pull request by welozano**
Closes #8

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -107,6 +107,13 @@
             <version>3.7.7</version>
             <scope>test</scope>
         </dependency>
+        <!-- To use univocity for csv parsing-->
+        <dependency>
+            <groupId>com.univocity</groupId>
+            <artifactId>univocity-parsers</artifactId>
+            <version>2.9.1</version>
+            <type>jar</type>
```
Comment (barbeau) [inline pom.xml:115]: I don't think you need `<type>jar</type>`

File: src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java [hunk 2]
```diff
@@ -68,7 +72,10 @@ public void processData() {
         // create csv file and add the header
         mCSVFileWriter.createHeader(TravelBehaviorRecord.CSV_HEADER);
 
-        if (mProgramOptions.getUserId() == null) {
+        if (mProgramOptions.getMultiUserId() != null) {
+            // analyze all data for a specific list of userIds
+            analyzeListOfUserIdTravelBehaviorData(mProgramOptions.getMultiUserId());
+        } else if (mProgramOptions.getUserId() == null) {
```
Comment (barbeau) [inline src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java:78]: I would invert the last two parts of this IF statement so the two exports for subsets of users comes first, and the full export of all users comes last.

File: src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java [hunk 3]
```diff
@@ -80,6 +87,20 @@ public void processData() {
...
+     * Retrieves a list of user ids in firebase and analyses each user's data one by one
+     */
+    private void analyzeListOfUserIdTravelBehaviorData(String pathToCSVFileListOfUserIds) {
+        List<String[]> allUserIds = mCSVFileReader.readUserList(pathToCSVFileListOfUserIds);
+        if (allUserIds.size() > 0) {
+            for(String[] userId : allUserIds){
+                processUserById(userId[0]);
+            }
+        } else {
+            System.err.println("The list of userId provided is empty or wrong formatted.");
```
Comment (barbeau) [inline src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java:100]:
```suggestion
            System.err.println("The list of userId provided is empty or incorrectly formatted.");
```

**Review by barbeau — COMMENTED** [review]
Thanks @welozano! Mostly LGTM, a few comments in-line.

File: pom.xml (hunk 1 above)
Comment (welozano) [reply pom.xml]: Resolved in the lasts commits.

File: src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java (hunk 2 above)
Comment (welozano) [reply src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java]: Resolved in the lasts commits.

File: src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java (hunk 3 above)
Comment (welozano) [reply src/main/java/edu/usf/cutr/tba/manager/TravelBehaviorDataAnalysisManager.java]: Changed in the lasts commits.

**Pull request merged**